src/kernel/context.h

```diff
@@ -19,7 +19,7 @@ namespace kernel {
 //! should be stored to std::unique_ptr members pointing to opaque types.
 struct Context {
     Context();
-    ~Context();
+    ~Context(); // NOLINT(performance-trivially-destructible)
```

Comment:
re: https://github.com/bitcoin/bitcoin/pull/29252#discussion_r1592642962

> There is already a comment in line 19, which explains why this is done ("should be stored to std::unique_ptr members pointing to opaque types.")
> 
> While there currently is no unique_ptr, it is possible that one will be re-added in the future?

Oh interesting. I was wondering why NOLINT was needed here but not in NodeContext, but I guess it is because NodeContext does contain unique_ptr members, and this doesn't have any. 

I think probably the simplest thing to do would be to wait until a destructor actually needs to be declared before adding one, instead of keeping a destructor that doesn't do anything anything anymore and needing to use NOLINT. It is possible the kernel context struct could have unique ptr members added in the future, but at this point that is looking less likely to happen (see https://github.com/bitcoin/bitcoin/pull/29252#discussion_r1592383067)